cmd/bootstrap/cmd/pull.go

```diff
@@ -62,15 +70,49 @@ func pull(cmd *cobra.Command, args []string) {
 	}
 	log.Info().Msgf("found %d files in google bucket", len(files))
 
+	sem := semaphore.NewWeighted(flagConcurrency)
+	wait := sync.WaitGroup{}
 	for _, file := range files {
-		if strings.Contains(file, "node-info.pub") {
-			fullOutpath := filepath.Join(flagOutdir, file)
-			log.Printf("downloading %s", file)
-
-			err = bucket.DownloadFile(ctx, client, fullOutpath, file)
-			if err != nil {
-				log.Error().Msgf("error trying download google bucket file: %v", err)
+		wait.Add(1)
+		go func(file gcs.GCSFile) {
+			_ = sem.Acquire(ctx, 1)
+			defer func() {
+				sem.Release(1)
+				wait.Done()
+			}()
+
+			if strings.Contains(file.Name, "node-info.pub") {
+				fullOutpath := filepath.Join(flagOutdir, file.Name)
+
+				fmd5 := CalcMd5(fullOutpath)
+				if bytes.Equal(fmd5, file.MD5) {
+					log.Printf("skipping %s", file)
+					return
+				}
+
+				log.Printf("downloading %s", file)
+				err = bucket.DownloadFile(ctx, client, fullOutpath, file.Name)
+				if err != nil {
+					log.Error().Msgf("error trying download google bucket file: %v", err)
+				}
 			}
-		}
+		}(file)
 	}
+
+	wait.Wait()
+}
+
+func CalcMd5(outpath string) []byte {
```

Comment:
It would be great to pull this out into `cmd/bootstrap/utils` so it isn't duplicated in `transit`

Reply:
moved CalcMd5 to cmd/bootstrap/utils